[flutter_local_notifications] Support bypassing DnD-settings on Android (#2562)
MaikuB/flutter_local_notifications: flutter_local_notifications/README.md

Developers should also be across Google's requirements on using full-screen intents. Please refer to their documentation [here](https://source.android.com/docs/core/permissions/fsi-limits) for more information. Should you app need request permissions, the `AndroidFlutterNotificationsPlugin` class exposes the `requestFullScreenIntentPermission()` method that can be used to do so.

> ### Bypassing dnd

Reviewer: ```suggestion
### Bypassing Do Not Disturb (DnD)
```

Suggested to make it easier to find as this will show up in the table of contents and explain the acronym. Android docs also refer to it without hyphens e.g. https://developer.android.com/develop/ui/views/notifications

> If your application needs the ability to send notifications that ignore the device's current Do-Not-Disturb settings, you must request these permissions from Android 6.0 onwards. You can do this using `AndroidFlutterNotificationsPlugin` by calling the `requestNotificationPolicyAccess()` method which will redirect the user to a settings page where your application may be explicitly whitelisted to bypass DnD.

Reviewer: ```suggestion
If your application needs the ability to send notifications that ignore the device's DnD mode, you must request these permissions from Android 6.0 onwards. You can do this using `AndroidFlutterNotificationsPlugin` by calling the `requestNotificationPolicyAccess()` method which will redirect the user to a settings page where your application may be explicitly whitelisted to bypass DnD.
```

- Uses the acronym
- settings -> mode . Using "mode" is more appropriate to describe that it's how the device is currently behaviour as its enter a "mode". It's also aligned with how Android describes it. Likely because settings is less clear as settings for DnD can be something set in advance and thereby makes using "current" redundant. An example of this is on setting up the bedtime schedule

For notifications to then actually ignore the DnD-status of a device, the channel must be created with `bypassDnd: true`, or the first notification on a channel that creates it must be sent with `channelBypassDnd: true`.

> **NOTE:** This does _not_ ignore the device's silent mode! Should you have a use-case where you must notify your users (for instance in an emergency), you might want to use a package like [`sound_mode`](https://pub.dev/packages/sound_mode) or write your own platform specific code to set the `RingerMode` of the device as well as change the notification stream's volume before and after the notification.

Reviewer: ```suggestion
**NOTE:** This does _not_ ignore the device's silent mode! Should you have a use case where you must notify your users (for instance in an emergency), you might want to use a package like [`sound_mode`](https://pub.dev/packages/sound_mode) or write your own platform specific code to set the `RingerMode` of the device as well as change the notification stream's volume before and after the notification.
```

use-case -> use case (hyphen isn't needed)

#### ProGuard rules

> For flutter_local_notification v19 and higher, the ProGuard rules are automatically provided by the GSON. The following documentation is for v18 and lower versions.

Reviewer: ```suggestion
For flutter_local_notifications v19 and higher, the ProGuard rules are automatically provided by the GSON. The following documentation is for v18 and lower versions.
```

Thanks for spotting the typo. Added a missing `s`
